newscout_web/advertising/views.py

```diff
@@ -65,20 +70,92 @@ def get(self, request, format=None, *args, **kwargs):
         return Response(create_response({"categories": categories.data, "campaigns": campaigns.data}))
 
 
+class GenericAdListAPIView(ListAPIView):
+    """
+    generic ad list api view
+    """
+    permission_classes = (AllowAny,)
+    pagination_class = PostpageNumberPagination
+    filter_backends = (filters.OrderingFilter,)
+    ordering = ('-id',)
+
+    def list(self, request, *args, **kwargs):
+        queryset = self.filter_queryset(self.get_queryset())
+
+        page = self.paginate_queryset(queryset)
+        if page is not None:
+            serializer = self.get_serializer(page, many=True)
+            if serializer.data:
+                paginated_response = self.get_paginated_response(serializer.data)
+                return Response(create_response(paginated_response.data))
+            else:
+                return Response(create_error_response({"Msg" : "Objects Doesn't Exist"}), status=400)
+
+        serializer = self.get_serializer(queryset, many=True)
+        return Response(create_response(serializer.data))
+
+
+class CampaignListView(GenericAdListAPIView):
+    """
+    campaign list view
+    """
+    serializer_class = CampaignSerializer
+
+    def get_queryset(self):
+        q = self.request.GET.get("q","")
+        queryset = Campaign.objects.all()
+
+        if q:
+            q_list = q.split(" ")
+            condition_1 = reduce(operator.or_, [Q(name__icontains=s) for s in q_list])
```

Comment:
Rename `condition_1` to `filter_condition`, because `Explicit is better than Implicit`

Reply:
fixed in commit id: 72f358238c9f9fc26ded0f3329cd92812c79660f